learn/work/fairness-readmission/index.qmd

```diff
@@ -0,0 +1,517 @@
+:::
+
+```{r}
+#| label: "plot-n-variables"
+#| message: false
+#| warning: false
+readmission_collapsed %>%
+  pivot_longer(starts_with("n_")) %>%
+  ggplot(aes(x = value)) +
+  geom_histogram() +
+  facet_wrap(vars(name), scales = "free_x")
+```
+
+For many patients, this was their first inpatient visit to this hospital system. During their stay, many patients receive 10-20 medications and experience several procedures.
+
+With a sense for the distributions of variables in this dataset, we can move on to splitting data up for modeling.
+
+## Resampling Data
+
+First, splitting data into training and testing:
+
+```{r}
+#| label: "split"
+set.seed(1)
+readmission_splits <- initial_split(readmission_collapsed)
+readmission_train <-  training(readmission_splits)
+readmission_test <-   testing(readmission_splits)
+```
+
+We'll set the `r format(nrow(readmission_test), big.mark = ",")`-row `readmission_test` test set to the side for the remainder of the analysis until we compute a final estimate of our performance on the chosen model. Splitting the `r format(nrow(readmission_train), big.mark = ",")` rows of the training data into 10 resamples:
+
+```{r}
+#| label: "resample"
+readmission_folds <- vfold_cv(readmission_train)
+
+readmission_folds
+```
+
+Each split contains an analysis and assessment set: one for model fitting and the other for evaluation. Averaging performance estimates across resamples will give us a sense for how well the model performs on data it hasn't yet seen.
+
+## Training and Evaluating Models
+
+We'll define a diverse set of models and pre-processing strategies and then evaluate them against our resamples.
+
+### Model Workflows
+
+We'll first define a basic recipe that first sets a factor level for missing values and then centers and scales numeric data:
+
+```{r}
+#| label: "recipe-basic"
+recipe_basic <-
+  recipe(readmitted ~ ., data = readmission) %>%
+  step_unknown(all_nominal_predictors()) %>%
+  step_YeoJohnson(all_numeric_predictors()) %>%
+  step_normalize(all_numeric_predictors()) %>%
+  step_dummy(all_nominal_predictors())
+```
+
+The other preprocessor that we'll try encodes the age as a numeric variable rather than the bins as in the source data:
+
+```{r}
+#| label: "recipe-age"
+# specify the age as the midpoint of the range
+# rather than as an interval, like in the source data
+# e.g. "[10-20]" -> 1 * 10 + 5 -> 15
+recipe_age <-
+  recipe(readmitted ~ ., data = readmission) %>%
+  step_mutate(
+    age_num = 
+      age %>% 
+      as.character() %>% 
+      substr(2, 2) %>% 
+      as.numeric() %>%
+      `*`(10) %>% `+`(5)
+  ) %>%
```

Comment:
The age thing right here isn't is a little much for a pipe in my opinion. I would pull it out as a helper function

```suggestion
# specify the age as the midpoint of the range
# rather than as an interval, like in the source data
# e.g. "[10-20]" -> 1 * 10 + 5 -> 15
get_age <- function(x) {
  res <- substr(x, 2, 2)
  res <- as.numeric(res)
  res * 10 + 5
}

recipe_age <-
  recipe(readmitted ~ ., data = readmission) %>%
  step_mutate(
    age_num = get_age(age)
  ) %>%
```

Reply:
OR you could be overly fancy and do

```r
step_mutate(
  age_num = as.numeric(age) * 10 - 5
)
```

Reply:
Heard! [115a87b](https://github.com/tidymodels/tidymodels.org/pull/52/commits/115a87bc8ac9c045789fa29970a5091889296d16).